src/hotspot/cpu/aarch64/interp_masm_aarch64.cpp

```diff
@@ -1783,6 +1783,8 @@ void InterpreterMacroAssembler::load_field_entry(Register cache, Register index,
   ldr(cache, Address(rcpool, ConstantPoolCache::field_entries_offset()));
   add(cache, cache, Array<ResolvedFieldEntry>::base_offset_in_bytes());
   lea(cache, Address(cache, index));
+  // Prevents stale data from being read after the bytecode is patched to the fast bytecode
+  membar(MacroAssembler::LoadLoad);
```

Comment:
if we put LoadLoad in here, maybe it is redundant for TemplateTable::patch_bytecode(..)
https://github.com/openjdk/jdk/pull/18477/files#diff-739c969c24180bfe592cd0a75940d3838503390d3f51c8362a56ec4903252b67R192

because there has a Load-acquire in L199.
can we change 
`load_field_entry(Register cache, Register index, int bcp_offset = 1) ` 
to 
`load_field_entry(Register cache, Register index, int bcp_offset = 1, bool needLoadLoad = 1) `?

then change L192 like this
<pre>
@@ -189,7 +189,7 @@ void TemplateTable::patch_bytecode(Bytecodes::Code bc, Register bc_reg,
       // additional, required work.
       assert(byte_no == f1_byte || byte_no == f2_byte, "byte_no out of range");
       assert(load_bc_into_bc_reg, "we use bc_reg as temp");
-      __ load_field_entry(temp_reg, bc_reg);
+      __ load_field_entry(temp_reg, bc_reg, 1 /*bcp_offset*/, false /*needLoadLoad*/);
       if (byte_no == f1_byte) {
         __ lea(temp_reg, Address(temp_reg, in_bytes(ResolvedFieldEntry::get_code_offset())));
       } else {
         __ lea(temp_reg, Address(temp_reg, in_bytes(ResolvedFieldEntry::put_code_offset())));
       }                                                                         
       // Load-acquire the bytecode to match store-release in ResolvedFieldEntry::fill_in()
       __ ldarb(temp_reg, temp_reg);
</pre>

Reply:
I believe patch_bytecode only needs a LoadStore between reading the cache bytecode and patching at bcp[0], so I would agree the LoadLoad in load_field_entry is not needed here.  But the reason is not because we have already load-acquire.  The reason is because there are not two loads that need ordering.  There is only a load and a store.  We can't replace the load-acquire with a LoadLoad, for example.

Reply:
The LoadLoad in load_field_entry is needed for the second thread that (t2-1) loads the patched bytecode, and then (t2-2) loads the pointer to the cpCache entry for field, then (t2-3) loads the values from the cpCache entry pointer.  The LoadLoad is between t2-2 and t2-3.

The ldarb in patch_bytecode does make the LoadLoad in load_field_entry redundant as pointed out by @sunny868 but seems harmless in terms of performance and correctness, ie. not needing a special parameter. 

The reason that it's in load_field_entry is that there are other callers who do not load acquire the get_code and put_code fields.  So it's safer and less error prone to forget with it in load_field_entry.